src/services/Content/ContentQueries.ts

```diff
@@ -443,3 +446,85 @@ export const useGetPackagesQuery = (
       },
     },
   );
+
+export const useIntrospectRepositoryMutate = (
+  queryClient: QueryClient,
+  page: number,
+  perPage: number,
+  filterData?: FilterData,
+  sortString?: string,
+) => {
+  // Below MUST match the "useContentList" key found above or updates will fail.
+  const contentListKeyArray = [
+    CONTENT_LIST_KEY,
+    page,
+    perPage,
+    sortString,
+    ...Object.values(filterData || {}),
+  ];
+  const { notify } = useNotification();
+  return useMutation(introspectRepository, {
+    onMutate: async (item: IntrospectRepositoryItem) => {
+    // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
+    await queryClient.cancelQueries(contentListKeyArray);
+    // Snapshot the previous value
+    const previousData: Partial<ContentListResponse> =
+      queryClient.getQueryData(contentListKeyArray) || {};
+
+      const previousStatus = previousData.data?.find((data) => item.uuid == data.uuid)?.status
+      const newData = cloneDeep(previousData)
+      if (previousStatus) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        newData.data.filter((data) => item.uuid == data.uuid).at(0).status = 'Retrying'
+      }
```

Comment:
is there a cleaner way to do this?

Reply:
Yes an no, I think there are alternatives, that you could use with variables. 
Instead of filter for example you could use findIndex > store that in a variable > and target the specific value in newData.data[foundIndex].status ="stuff".

Also, anytime you use a variable chain on a filter, you should likely safe those values with ?. 
And then make the assignment conditional.

Reply:
Why was "Retrying" necessary here. I'd expect Pending to work just as well.

Reply:
I made the Retrying status because Pending implied that no packages were available (since the repository had never been introspected). With Retrying, packages could be available already. And you can see that in the UI if you have a repository and "retry", the package count doesn't switch to "N/A".

Reply:
I would still disable the package count while introspecting in any case, as that data may change. UNLESS, when your retrying and successful you somehow invalidate that package list cache.

Reply:
I made it N/A for retrying. It may have been necessary for me to add a new status...I'm trying to think if there's another reason for an entirely different status. If it not, it probably makes more sense to just think about the Pending status differently.

Reply:
And in response to "is there a cleaner way to do this". I can't really figure out another way to do it where typescript doesn't also complain. But if you have a specific way of doing it that you like that you like more, feel free to show me how :).

Reply:
I'll tinker on monday when I do another review.